source/guides/best-practices.md

```diff
@@ -31,10 +31,8 @@ rec {
 { a = 1; b = 3; }
 ```
 
-There are a couple of pitfalls:
-
-- It's possible to introduce a hard to debug error `infinite recursion` when shadowing a name, the simplest example being `let b = 1; a = rec { b = b; }; in a`.
-- Combining with overriding logic such as the [`overrideAttrs`](https://nixos.org/manual/nixpkgs/stable/#sec-pkg-overrideAttrs) function in {term}`Nixpkgs` has a surprising behaviour of not overriding every reference.
+A common pitfall is to introduce a hard to debug error `infinite recursion` when shadowing a name.
+The simplest example for this is: `let a = 1; in rec { a = a; }`.
```

Comment:
```suggestion
The simplest example for this is: `let a = 1; in rec { a = a; }`
```

With a colon in between we probably don't need the full stop, as that tends to interfere with reading the expression.

Alternatively, but that could give it too much emphasis right before the tip:

````suggestion
The simplest example for this is:

```{code-block} nix
let a = 1; in rec { a = a; }
```
````

We may want to add `<details>` because it may not be apparent what's trivially weird about it:

> One might expect that the value assigned in the attribute set is taken from the surrounding `let` binding, but names from the `rec` scope take precedence.